### PR TITLE
[CONTENTS] Cards

### PR DESCRIPTION
**Pull request by mikorosales**
## Issue Link
[LMS-307](https://framgiaph.backlog.com/view/LMS-307)

## Definition of Done
Should have a card component under shared > components

**For your convenience you can access this link**
- http://localhost:3000/demo/components#card

## Steps to reproduce
- import the card component in the page you want it to be utilized
- add height to the card component since it passes height as props

```
import Card from '@/src/shared/components/Card';

<Card />
```

## Affected Components / Functionalities / Page
n/a

## Test Cases
_will update soon..._

## Notes
n/a

## Screenshots
image
image

### CODE REVIEW

File: client/src/shared/components/Card/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+interface CardProps {
+  height: string
+  title: string
+  content: string
```
Comment (escbooster12-dev) [inline client/src/shared/components/Card/index.tsx:4]: - card should be dynamic as possible. By just passing `string` would not be enough.
- it would be better passing a `children` instead

```suggestion
  children: React.ReactNode
```

```
// usage

<Card>
	<main>
		Hello, This is the content!
	</main>
</Card>
```

File: client/src/shared/components/Card/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        </div>
+      </div>
+    </div>
+  );
+};
+
+Card.defaultProps = {
+  height: 'h-30',
+  title: 'Card Title',
+  content: 'Lorem Ipsum...'
```
Comment (escbooster12-dev) [inline client/src/shared/components/Card/index.tsx:26]: - do not set any specific default values in the props. Please set it as empty.
 ```suggestion
        className={`bg-white rounded-lg overflow-hidden shadow-lg h-30 ${height} w-auto`}
      >
        <div className="px-6 py-4">
          <h3 className="text-lg font-medium text-gray-900 mb-2">{title}</h3>
          <p className="text-gray-600">{content}</p>
        </div>
      </div>
    </div>
  );
};

Card.defaultProps = {
  height: '',
  title: '',
  children: React.ReactNode
```

File: client/src/pages/demo/components/index.tsx [hunk 3]
```diff
@@ -32,6 +33,19 @@ const DemoComponent: React.FunctionComponent = () => {
         <br />
         <hr className="w-3/4 mx-auto" />
       </div>
+
+      <div className="p-4 border -b border-black">
+        <Card>
+          <main>Content goes here</main>
+        </Card>
+        <div className="mt-[5px]">
+          <h1>Props: Card</h1>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:42]: - should be an anchor tag. please follow here https://github.com/framgia/sph-lms/pull/10#discussion_r1123993692

File: client/src/pages/demo/components/index.tsx [hunk 4]
```diff
@@ -32,6 +33,19 @@ const DemoComponent: React.FunctionComponent = () => {
         <br />
         <hr className="w-3/4 mx-auto" />
       </div>
+
+      <div className="p-4 border -b border-black">
+        <Card>
+          <main>Content goes here</main>
+        </Card>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:40]: - can you demonstrate as well other props like height and title :bow:

File: client/styles/globals.css [hunk 5]
```diff
@@ -1,112 +1,3 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
-
-
-:root {
```
Comment (ragg09-sunasterisk) [inline client/styles/globals.css:6]: No need to remove this, please undo the changes in this file

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- Card can be seen in the page and can be accessed thru http://localhost:3000/demo/components#card
- Card component is presented under shared > components
- No ESLint warnings or errors.

Comments: 5
Test time: 15 mins
Iterations: 2

File: client/styles/globals.css [hunk 6]
```diff
@@ -1,3 +1,113 @@
...
+    color: inherit;
+    text-decoration: none;
+  }
+  
+  @media (prefers-color-scheme: dark) {
+    html {
+      color-scheme: dark;
+    }
+  }
+
```
Comment (escbooster12-dev) [inline client/styles/globals.css:113]: - these code has been removed in the latest master branch. Please undo these changes.

File: client/styles/globals.css [hunk 7]
```diff
@@ -1,3 +1,4 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
```
Comment (escbooster12-dev) [inline client/styles/globals.css:4]: - missing newline
image

**Review by escbooster12-dev — APPROVED** [review]
Comments: 7
Iteration: 5
